src/mono/browser/runtime/assets.ts

```diff
@@ -104,6 +106,23 @@ export async function instantiate_symbols_asset(pendingAsset: AssetEntryInternal
     }
 }
 
+export async function instantiate_static_json_asset(pendingAsset: AssetEntryInternal): Promise<void> {
+    try {
+        const response = await pendingAsset.pendingDownloadInternal!.response;
+        const json = await response.json();
+        
+        switch (pendingAsset.name) {
+            case "segmentation-rules.json":
```

Comment:
avoid matching file names, let's use more specific asset.behavior name

Reply:
Right, my thinking behind this was to prepare ground if in the future we would need more "static-json" resources to be loaded. I can make this specific just to segmentation rules if that would be better. (related to https://github.com/dotnet/runtime/pull/95583#discussion_r1442915673)